### PR TITLE
completed assessment

### PR DESCRIPTION
**Pull request by crgjr66**
didn't know alot about the items for next week, so lots to learn for me

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -12,27 +12,47 @@ var num2 = 0
...
+    return `${number} is divisible by three`
+  } else {
+    return `${number} is not divisible by three`
+  }
+}
 
-
+console.log(divBy3(num1));
+console.log(divBy3(num2));
+console.log(divBy3(num3));
```
Comment (andeeisaacs) [inline code-challenges.js:25]: Nice and clean. Good work!

File: code-challenges.js [hunk 2]
```diff
@@ -12,27 +12,47 @@ var num2 = 0
...
 
 var randomNouns = ["streetlamp", "potato", "teeth", "conclusion", "nephew", "temperature", "database"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew", "Temperature", "Database"]
 
+const firstLetterCap = (array) => {
+  return array.map(value => value.charAt(0).toUpperCase() + value.slice(1))
+}
 
-
+console.log(firstLetterCap(randomNouns));
```
Comment (andeeisaacs) [inline code-challenges.js:37]: 🤩 Amazing, all in one return! Beautiful!

File: code-challenges.js [hunk 3]
```diff
@@ -12,27 +12,47 @@ var num2 = 0
...
 
+const onlyNums = (array) => {
+  // first find/filter the numbers
+  let numArray = []
+  numArray = array.filter(value => (typeof value === "number"))
 
+  // then sort the numbers from least to greatest
+  // by sorting based on the difference between the 2 items being compared
+  return numArray.sort((a,b)=>a-b);
+}
```
Comment (andeeisaacs) [inline code-challenges.js:53]: Excellent working code!!!

Nothing wrong at all with how you wrote this, its fantastic, but I do want to show you another way we can do that single line return like you used above in a previous challenge-

`const onlyNums = (array) => {`
 ` return array.filter(value => (typeof value === "number")).sort((a,b)=>a-b);`
`}`

What we see here is exactly the same thing you had, but all in one return and chaining on the methods in their order of operation. 

Just wanted to show you another way we can keep our use of extra variables and returns down.

File: code-challenges.js [hunk 4]
```diff
@@ -41,25 +61,52 @@ var vowelTester1 = "learn"
...
+  for(let i = 0; i < string.length; i++) {
+    if (vowels.includes(string[i])) {
+      // return the index as soon as you find it
+      return i;
+    }
+  }
+}
 
+console.log(firstVowel(vowelTester1));
+console.log(firstVowel(vowelTester2));
```
Comment (andeeisaacs) [inline code-challenges.js:78]: Fantastic! I would be interested in seeing how we could refactor this out of a for loop and utilizing a higher-order function instead.

File: code-challenges.js [hunk 5]
```diff
@@ -41,25 +61,52 @@ var vowelTester1 = "learn"
...
-// Expected output: 27
+ console.log(calculator(3, "*", 9))
+ // Expected output: 27
 
-// console.log(calculator(16, "+", 3))
+ console.log(calculator(16, "+", 3))
 // Expected output: 19
 
-// console.log(calculator(89, "/", 0))
+ console.log(calculator(89, "/", 0))
```
Comment (andeeisaacs) [inline code-challenges.js:109]: Excellent work!!!

File: code-challenges.js [hunk 6]
```diff
@@ -69,7 +116,45 @@ var vowelTester2 = "throw"
...
+      if (currentSite[i] === '.' && currentSite[i + 1] === 'i') {
+        // assume we found an .io website
+        ioSitesArray.push(currentSite)
+        break
+      } // end if
+    } // end for
+  }); // end .map
+
+  return ioSitesArray
+}
```
Comment (andeeisaacs) [inline code-challenges.js:140]: This is great, but I would look into how Map already performs a loop for us, thus rendering the nested loop unnecessary.

File: code-challenges.js [hunk 7]
```diff
@@ -69,7 +116,45 @@ var vowelTester2 = "throw"
...
+  // need an array for the values
+  let numberArray = [0]
+  // loop until we reach our number
+  while(count++ < number) {
+    // keep track of the numbers
+    numberArray.push(count);
+  }
+  return numberArray;
+}
+
```
Comment (andeeisaacs) [inline code-challenges.js:159]: This totally works great! I'd maybe refacort this to see how we can do it without having to push into an array.

File: interview-questions.md [hunk 8]
```diff
@@ -7,90 +7,111 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. Why would you use the method `super()`?
 
-  Your answer:
+  Your answer: super() is used to gain access to functions in a parent class
```
Comment (andeeisaacs) [inline interview-questions.md:10]: 👍

File: interview-questions.md [hunk 9]
```diff
@@ -7,90 +7,111 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: By calling the super() method in the constructor method, we call the parent's constructor method and gets access to the parent's properties and methods.
 
 
 
 2. What is the DOM? What is the virtual DOM?
 
-  Your answer:
+  Your answer: Document Object Model, and virtual DOM sits somewhere between our code and the real DOM, but I'm not really confident what they are
```
Comment (andeeisaacs) [inline interview-questions.md:18]: No worries! You got it right! The virtual DOM is React specific, allowing us and our users to make updates and changes to our DOM without refreshing the entire page.

File: interview-questions.md [hunk 10]
```diff
@@ -7,90 +7,111 @@ Then, research the question to expand on your answer. Even if you feel you have
...
+  Core DOM - standard model for all document types
+  XML DOM - standard model for XML documents
+  HTML DOM - standard model for HTML documents
 
 
 
 3. What is a higher-order function?
 
-  Your answer:
+  Your answer: They are functions that take other functions as arguments.
```
Comment (andeeisaacs) [inline interview-questions.md:37]: Absolutely. I would maybe want to expand on this a bit more.

File: interview-questions.md [hunk 11]
```diff
@@ -7,90 +7,111 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: Higher-order functions are functions that take other functions as arguments or return functions as their results. Taking an other function as an argument is often referred as a callback function, because it is called back by the higher-order function. The term comes from mathematics, where the distinction between functions and other values is taken more seriously. Higher-order functions allow us to abstract over actions, not just values.
 
 
 
 4. What is JSX?
 
-  Your answer:
+  Your answer: Javascript XML, allowing us to write HTML in our Javascript.
```
Comment (andeeisaacs) [inline interview-questions.md:45]: 💯

File: interview-questions.md [hunk 12]
```diff
@@ -7,90 +7,111 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: JSX stands for JavaScript XML. JSX allows us to write HTML in React. JSX makes it easier to write and add HTML in React. JSX is a preprocessor step that adds XML syntax to JavaScript. Just like XML, JSX tags have a tag name, attributes, and children. If an attribute value is enclosed in quotes, the value is a string. Otherwise, wrap the value in braces and the value is the enclosed JavaScript expression.
 
 
 
 5. What is React?
 
-  Your answer:
+  Your answer: React is a Javascript library and is component based; created by Facebook.
```
Comment (andeeisaacs) [inline interview-questions.md:53]: Absolutely. I would want to expand on it and mention that its specifically a Library allowing us to create dynamic, component based, reactive front-end applications.

File: interview-questions.md [hunk 13]
```diff
@@ -7,90 +7,111 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: A JavaScript library for building user interfaces. React makes it painless to create interactive UIs, Design simple views for each state in your application, and React will efficiently update and render just the right components when your data changes. React.js is an open-source JavaScript library that is used for building user interfaces specifically for single-page applications. React allows developers to create large web applications that can change data, without reloading the page. The main purpose of React is to be fast, scalable, and simple
 
 
 
 6. What is yarn? What file(s) are updated when you run `$ yarn`?
 
-  Your answer:
+  Your answer: Yarn is a javascript package manager that automates, installs, updates, and configures software packages from a global registry.  I don't exactly know what files are updated when you run '$ yarn'.
```
Comment (andeeisaacs) [inline interview-questions.md:61]: You got it right! We could say that when we run Yarn, it updates our yarn.lock and package files for us.

File: interview-questions.md [hunk 14]
```diff
@@ -7,90 +7,111 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: Yarn was developed by Facebook in attempt to resolve some of npm's shortcomings. Yarn isn't technically a replacement for npm since it relies on modules from the npm registry. Yarn is a new installer that still relies upon the same npm structure. The registry itself hasn't changed, but the installation method is different. Since Yarn gives you access to the same packages as npm, moving from npm to Yarn doesn't require you to make any changes to your workflow.  Yarn uses a lock file to track dependencies and file versions.
 
 
 
 7. What is localhost:3000? How do we use localhost:3000 in the development process?
 
-  Your answer:
+  Your answer: It is Port 3000 on 'this' computer.  It is used as a server to output our HTML and Javascript React projects, allowing us to see our output and debug our code as we are developing.
```
Comment (andeeisaacs) [inline interview-questions.md:69]: Excellent!

File: interview-questions.md [hunk 15]
```diff
@@ -7,90 +7,111 @@ Then, research the question to expand on your answer. Even if you feel you have
...
 
-  Researched answer:
+  Researched answer: Iterable objects is a generalization of arrays. That’s a concept that allows us to make any object useable in a for..of loop. Of course, Arrays are iterable. But there are many other built-in objects, that are iterable as well. For instance, strings are also iterable. An iterable object is any object that returns a function that produces an Iterator for its Symbol. ... You can loop over all values in an iterable object by using a for (var value of iterable) { } loop.
 
 
 
 9. STRETCH: What is hoisting in JavaScript?
 
-  Your answer:
+  Your answer: I don't know
```
Comment (andeeisaacs) [inline interview-questions.md:85]: No worries. I know you got it with your researched answer :)

**Review by andeeisaacs — APPROVED** [review]
Chuck! You did great this assessment! 

My only piece of advice I would give you is to start to see what other methods we could be using other than a for loop. I would also advise to look into .map and see how it does iteration for you as well. 

Great work! Please merge to master!

**Comment by crgjr66** [comment]
for loops and while loops feel natural because of the logical thought process, but I will make an effort to always use high order functions and compress the code as much as possible

**Pull request merged**